awx/conf/migrations/_create_dab_ldap_config.py

```diff
@@ -0,0 +1,204 @@
+                                organization[user_type] = [organization[user_type]]
+
+                            triggers = {"groups": {"has_or": organization[user_type]}}
+
+                        map_config = {
+                            "authenticator": authenticator_id,
+                            "revoke": organization.get(f'remove_{user_type}', False),
+                            "map_type": "team",
+                            "team": f"Organization {user_type.title()}",
+                            "organization": organization_name,
+                            "triggers": triggers,
+                        }
+                        map_configs.append(map_config)
+
+        if "TEAM_MAP" in awx_settings:
+            for team_name in awx_settings.get("TEAM_MAP").keys():
+                team = awx_settings.get("TEAM_MAP")[team_name]
+                if team['users'] is None:
+                    continue
+                if team['users'] is False:
+                    triggers = {"never": {}}
+                elif team['users'] is True:
+                    triggers = {"always": {}}
+                else:
+                    if type(team['users']) is str:
+                        team['users'] = [team['users']]
+
+                    triggers = {"groups": {"has_or": team['users']}}
+
+                map_config = {
+                    "authenticator": authenticator_id,
+                    "revoke": team.get('remove', False),
+                    "map_type": "team",
+                    "team": team_name,
+                    "organization": team.get('organization', 'You have a team with no organization'),
+                    "triggers": triggers,
+                }
+                map_configs.append(map_config)
+
+        require_group = awx_settings.get("REQUIRE_GROUP", None)
+        if require_group:
+            map_config = {
+                "authenticator": authenticator_id,
+                "revoke": False,
+                "map_type": "allow",
+                "team": None,
+                "organization": None,
+                "triggers": {"groups": {"has_and": [require_group]}},
+            }
+            map_configs.append(map_config)
+
+        deny_group = awx_settings.get("DENY_GROUP", None)
+        if deny_group:
+            map_config = {
+                "authenticator": authenticator_id,
+                "revoke": False,
+                "map_type": "allow",
+                "team": None,
+                "organization": None,
+                "triggers": {"groups": {"has_not": [require_group]}},
+            }
+            map_configs.append(map_config)
+
+    return map_configs
+
+
+def create_ldap_auth_and_authmap(apps, schema_editor):
+    logger.info('Creating django-ansible-base LDAP authenticators and authenticator maps...')
+    awx_ldap_group_settings = get_awx_ldap_settings()
+    for awx_ldap_name, awx_ldap_settings in enumerate(awx_ldap_group_settings.values()):
```

Comment:
This seems not quite right?  Plain Python dicts are not in a guaranteed order.

Anyway, if I'm following what is going on with `get_awx_ldap_settings()`, aren't the keys of the outer dict the integers that you want for `awx_ldap_name`?  If so, iterating over `awx_ldap_group_settings.items()` would make more sense.